### PR TITLE
Move AppConfiguration to main

### PR DESCRIPTION
**Pull request by BethanyZhou**
## Description

Move https://github.com/Azure/azure-powershell/pull/17531 to main

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by BethanyZhou** [comment]
@lijinpei2008 , please add change log message for this PR

**Comment by lijinpei2008** [comment]
> @lijinpei2008 , please add change log message for this PR

Done.

File: src/AppConfiguration/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Add parameter "publicNetworkAccess" in {New/Update}-AzAppConfigurationStore
```
Comment (BethanyZhou) [inline src/AppConfiguration/ChangeLog.md:21]:
```suggestion
* Add parameter "publicNetworkAccess" in {New/Update}-AzAppConfigurationStore
```
```suggestion
* Added parameter "publicNetworkAccess" in `New-AzAppConfigurationStore` and `Update-AzAppConfigurationStore`
```

File: src/AppConfiguration/ChangeLog.md [hunk 2]
```diff
@@ -18,7 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
-* Add parameter "publicNetworkAccess" in {New/Update}-AzAppConfigurationStore
+* Added parameter "publicNetworkAccess" in `New-AzAppConfigurationStore` and `Update-AzAppConfigurationStore`
```
Comment (isra-fel) [inline src/AppConfiguration/ChangeLog.md:21]: publicNetworkAccess -> PublicNetworkAccess

File: src/AppConfiguration/ChangeLog.md [hunk 3]
```diff
@@ -18,6 +18,7 @@
         - Additional information about change #1
 -->
 ## Upcoming Release
+* Added parameter "publicNetworkAccess" in `New-AzAppConfigurationStore` and `Update-AzAppConfigurationStore`
```
Comment (BethanyZhou) [inline src/AppConfiguration/ChangeLog.md:21]:
```suggestion
* Added parameter "PublicNetworkAccess" in `New-AzAppConfigurationStore` and `Update-AzAppConfigurationStore`
```

**Review by isra-fel — APPROVED** [review]
LGTM

**Pull request merged**